src/parameterizations/vertical/MOM_diabatic_driver.F90

```diff
@@ -287,8 +292,37 @@ subroutine diabatic(u, v, h, tv, Hml, fluxes, visc, ADp, CDp, dt, Time_end, &
   integer :: i, j, k, m, is, ie, js, je, nz
   logical :: showCallTree ! If true, show the call tree
 
+  real, dimension(SZI_(G),SZJ_(G))            :: sppt_wts
+  real, dimension(SZI_(G),SZJ_(G),2)          :: t_rp
+#ifdef UFS
+  real, dimension(SZI_(G),SZJ_(G),SZK_(G))   :: h_in
+  real, dimension(SZI_(G),SZJ_(G),SZK_(G))   :: t_in        !< thickness [H ~> m or kg m-2]
+  real, dimension(SZI_(G),SZJ_(G),SZK_(G))   :: s_in        !< thickness [H ~> m or kg m-2]
+  real :: t_tend,s_tend,h_tend                  ! holder for tendencey needed for SPPT
+  real :: t_pert,s_pert,h_pert                  ! holder for tendencey needed for SPPT
+#endif
+
   if (G%ke == 1) return
 
+#ifdef UFS
+   ! save  copy of the date for SPPT
+  if (CS%do_sppt) then
+     h_in=h
+     t_in=tv%T
+     s_in=tv%S
+   endif
```

Comment:
shouldn't the rest of this, such as the call run_stochastic all be behind the do_sppt?

Reply:
It would have to be do_sppt .or. do_epbl, which I could add.

Even if I have a do_sppt and do_epbl logic check before calling the subroutine,  the subroutine will still have to be present when compiling other models such as GFDL or NCAR's version.

Reply:
But you shouldn't call the subroutine if it wasn't do_sppt .or. do_epbl, correct?  Right now if you have the ifdef, you're calling the stochastic physics whether you are turning stochastic on or not?

Reply:
correct. and the logic is handled inside, but I will change it so the model only calls run_stochastic_physics if do_sppt and do_epbl are true